locallib.php

```diff
@@ -799,3 +810,56 @@ function zoom_get_alternative_host_array_from_string($alternativehoststring) {
     $alternativehostarray = array_filter(explode(',', $alternativehoststring));
     return $alternativehostarray;
 }
+
+/**
+ * Get all meeting records
+ *
+ * @return array All zoom meetings stored in the database.
+ */
+function zoom_get_all_meeting_records() {
+    global $DB;
+
+    $meetings = [];
+    $records = $DB->get_records('zoom');
+    foreach ($records as $record) {
+        $meetings[$record->meeting_id] = $record;
+    }
+
+    return $meetings;
+}
+
+/**
+ * Get all recordings for a particular meeting
+ *
+ * @param int $zoomid The id of the zoom meeting
+ *
+ * @return array All the recordings for the zoom meeting
+ */
+function zoom_get_meeting_recordings($zoomid) {
+    global $DB;
+
+    $recordings = $DB->get_records('zoom_meeting_recordings', ['zoomid' => $zoomid]);
+    foreach ($recordings as $key => $recording) {
+        unset($recordings[$key]);
+        $recordings[$recording->zoomrecordingid] = $recording;
+    }
+    return $recordings;
+}
+
+/**
+ * Get all meeting recordings grouped together
+ *
+ * @param int $zoomid The id of the zoom meeting
+ *
+ * @return array All recordings for the zoom meeting grouped together
+ */
+function zoom_get_meeting_recordings_grouped($zoomid) {
+    global $DB;
+
+    $records = $DB->get_records('zoom_meeting_recordings', ['zoomid' => $zoomid], 'timecreated ASC');
```

Comment:
Wouldn't we care more about the recording start time?
```suggestion
    $records = $DB->get_records('zoom_meeting_recordings', ['zoomid' => $zoomid], 'recordingstart ASC');
```